include/deal.II/base/function.templates.h

```diff
@@ -297,240 +297,200 @@ Function<dim, Number>::memory_consumption () const
-  :
-  ZeroFunction<dim, Number> (values.size()),
-  function_value_vector (values)
-{}
-
-
-template <int dim, typename Number>
-ConstantFunction<dim, Number>::
-ConstantFunction (const Vector<Number> &values)
-  :
-  ZeroFunction<dim, Number> (values.size()),
-  function_value_vector (values.size())
-{
-  Assert (values.size() == function_value_vector.size(),
-          ExcDimensionMismatch (values.size(), function_value_vector.size()));
-  std::copy (values.begin(),values.end(),function_value_vector.begin());
-}
-
+namespace Functions
+{
+
+  template <int dim, typename Number>
+  ConstantFunction<dim, Number>::ConstantFunction (const Number value,
+                                                   const unsigned int n_components)
+    :
+    Function<dim, Number> (n_components),
+    function_value_vector (n_components, value)
+  {}
+
+  template <int dim, typename Number>
+  ConstantFunction<dim, Number>::
+  ConstantFunction (const std::vector<Number> &values)
+    :
+    Function<dim, Number> (values.size()),
+    function_value_vector (values)
+  {}
+
+
+  template <int dim, typename Number>
+  ConstantFunction<dim, Number>::
+  ConstantFunction (const Vector<Number> &values)
+    :
+    Function<dim, Number> (values.size()),
+    function_value_vector (values.size())
+  {
+    Assert (values.size() == function_value_vector.size(),
+            ExcDimensionMismatch (values.size(), function_value_vector.size()));
+    std::copy (values.begin(),values.end(),function_value_vector.begin());
+  }
 
-template <int dim, typename Number>
-ConstantFunction<dim, Number>::
-ConstantFunction (const Number *begin_ptr, const unsigned int n_components)
-  :
-  ZeroFunction<dim, Number> (n_components),
-  function_value_vector (n_components)
-{
-  Assert (begin_ptr != nullptr, ExcMessage ("Null pointer encountered!"));
-  std::copy (begin_ptr, begin_ptr+n_components, function_value_vector.begin());
-}
+
+  template <int dim, typename Number>
+  ConstantFunction<dim, Number>::
+  ConstantFunction (const Number *begin_ptr, const unsigned int n_components)
+    :
+    Function<dim, Number> (n_components),
+    function_value_vector (n_components)
+  {
+    Assert (begin_ptr != nullptr, ExcMessage ("Null pointer encountered!"));
+    std::copy (begin_ptr, begin_ptr+n_components, function_value_vector.begin());
+  }
 
 
 
-template <int dim, typename Number>
-ConstantFunction<dim, Number>::~ConstantFunction ()
-{
-  function_value_vector.clear();
-}
-
-
-template <int dim, typename Number>
-Number ConstantFunction<dim, Number>::value (const Point<dim> &,
-                                             const unsigned int component) const
-{
-  Assert (component < this->n_components,
-          ExcIndexRange (component, 0, this->n_components));
-  return function_value_vector[component];
-}
+  template <int dim, typename Number>
+  ConstantFunction<dim, Number>::~ConstantFunction ()
+  {
+    function_value_vector.clear();
```

Comment:
this automatically happens whenever `function_value_vector` is destroyed. In other words, you don't need this line, and then you don't need the entire destructor either -- just remove its implementation and declaration.